src/main/java/seedu/address/logic/commands/OpenCommand.java

```diff
@@ -0,0 +1,218 @@
+
+    private final Index index;
+    private final UpdateRideDescriptor openRideDescriptor;
+    /**
+     * @param index of the ride in the filtered ride list to open
+     */
+    public OpenCommand(Index index) {
+        requireNonNull(index);
+
+        this.index = index;
+        this.openRideDescriptor = new UpdateRideDescriptor();
+    }
+
+    @Override
+    public CommandResult execute(Model model, CommandHistory history) throws CommandException {
+        requireNonNull(model);
+        List<Ride> lastShownList = model.getFilteredRideList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_RIDE_DISPLAYED_INDEX);
+        }
+
+        Ride rideToOpen = lastShownList.get(index.getZeroBased());
+        Ride editedRide = createUpdatedRide(rideToOpen, openRideDescriptor);
+
+        if (rideToOpen.isSameRide(editedRide) && rideToOpen.equals(editedRide)) {
+            throw new CommandException(MESSAGE_DUPLICATE_RIDE);
+        }
+
+        model.updatePerson(rideToOpen, editedRide);
+        model.updateFilteredRideList(PREDICATE_SHOW_ALL_RIDES);
+        model.commitAddressBook();
+        return new CommandResult(String.format(MESSAGE_OPEN_RIDE_SUCCESS, editedRide));
+    }
+
+    /**
+     * Creates and returns a {@code Ride} with the details of {@code rideToOpen}
+     * edited with {@code openRideDescriptor}.
+     */
+    private static Ride createUpdatedRide(Ride rideToOpen, UpdateRideDescriptor openRideDescriptor) {
+        assert rideToOpen != null;
+
+        Name updatedName = openRideDescriptor.getName().orElse(rideToOpen.getName());
+        Maintenance updatedMaintenance =
+                openRideDescriptor.getMaintenance().orElse(rideToOpen.getDaysSinceMaintenance());
+        WaitTime updatedWaitTime = openRideDescriptor.getWaitTime().orElse(rideToOpen.getWaitingTime());
+        Address updatedAddress = openRideDescriptor.getAddress().orElse(rideToOpen.getAddress());
+        Set<Tag> updatedTags = openRideDescriptor.getTags().orElse(rideToOpen.getTags());
+        Status updatedStatus = Status.OPEN;
+
+        return new Ride(updatedName, updatedMaintenance, updatedWaitTime, updatedAddress, updatedTags, updatedStatus);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof OpenCommand)) {
+            return false;
+        }
+
+        // state check
+        OpenCommand e = (OpenCommand) other;
+        return index.equals(e.index)
+                && openRideDescriptor.equals(e.openRideDescriptor);
+    }
+
+    /**
+     * Stores the details to edit the ride with. Each non-empty field value will replace the
+     * corresponding field value of the ride.
+     */
+    public static class UpdateRideDescriptor {
```

Comment:
Maybe can try to make this a separate java class file, especially when it is public. Also, the fields have a lot in common compared to `Ride`. Can this extend `Ride` maybe?